test/Altinn.Notifications.Tests/Notifications.Integrations/SendCondition/SendConditionClientTests.cs

```diff
@@ -0,0 +1,145 @@
+                    },
+                    "oknobody" => new HttpResponseMessage(HttpStatusCode.OK)
+                    {
+                        Content = new StringContent("{\"isValidJson\": true}")
+                    },
+                    "invalidbody" => new HttpResponseMessage(HttpStatusCode.OK)
+                    {
+                        Content = new StringContent("This is not a valid JSON")
+                    },
+                    "badrequest" => new HttpResponseMessage(HttpStatusCode.BadRequest)
+                    {
+                        Content = new StringContent("Bad request")
+                    },
+                    _ => new HttpResponseMessage(HttpStatusCode.OK)
+                    {
+                        Content = new StringContent(JsonSerializer.Serialize(new SendConditionResponse { SendNotification = true }, _serializerOptions))
+                    },
+                };
+            });
+
+            _sendConditionClient = new SendConditionClient(new HttpClient(messageHandler));
+        }
+
+        [Fact]
+        public async Task CheckSendCondition_SendConditionTrue_ReturnsTrue()
+        {
+            // Act
+            Result<bool, ConditionClientError> result = await _sendConditionClient.CheckSendCondition(new Uri("http://test.com?desiredResponse=true"));
+
+            // Assert
+            result.Match(
+                sendNotification =>
+                {
+                    Assert.True(sendNotification);
+                    return true;
+                },
+                actuallError => throw new Exception("No error value should be returned if send condition response is true"));
+        }
+
+        [Fact]
+        public async Task CheckSendCondition_SendContitionFalse_ReturnsFalse()
+        {
+            // Act
+            Result<bool, ConditionClientError> result = await _sendConditionClient.CheckSendCondition(new Uri("http://test.com?desiredResponse=false"));
+
+            // Assert
+            result.Match(
+                sendNotification =>
+                {
+                    Assert.False(sendNotification);
+                    return true;
+                },
+                actuallError => throw new Exception("No error value should be returned if send condition response is false"));
+        }
+
+        [Fact]
+        public async Task CheckSendCondition_InvalidJson_ReturnsClientError()
+        {
+            // Act
+            Result<bool, ConditionClientError> result = await _sendConditionClient.CheckSendCondition(new Uri("http://test.com?desiredResponse=invalidbody"));
+
+            // Assert
+            result.Match(
+                sendNotification => throw new Exception("No success value should be returned if json deserialization fails"),
+                actuallError =>
+                {
+                    Assert.False(string.IsNullOrEmpty(actuallError?.Message));
```

Comment:
Should be unnecessary to check if `actualError` is null. The unit tests of `Result<T>` should check the `Result<T>` implementation and ensure that it never match on `actualError` if it's null.

This is repeated a few times in other tests as well.